src/wp-includes/blocks.php

```diff
@@ -804,7 +803,14 @@ function make_before_block_visitor( $context ) {
 			// Candidate for first-child insertion.
 			$relative_position  = 'first_child';
 			$anchor_block_type  = $parent['blockName'];
-			$hooked_block_types = array_keys( get_hooked_blocks( $anchor_block_type, $relative_position ) );
+			$hooked_block_types = get_hooked_blocks( $anchor_block_type );
+
+			if ( array_key_exists( $relative_position, $hooked_block_types ) ) {
+				$hooked_block_types = $hooked_block_types[ $relative_position ];
+			} else {
+				$hooked_block_types = array();
+			}
```

Comment:
These extra checks are probably the biggest downside of the suggested change (needed for each position; see below).

The alternative would be to make `get_hooked_blocks` create empty arrays for all positions, but I didn't like that option so much either; one drawback would be that we'd need to give `get_hooked_blocks` a list of possible relative positions (which we'd also need to update if we introduce new ones); whereas with the current approach, we can keep `get_hooked_blocks` ignorant of what relative positions exist, and let it create arrays for whatever it finds.